connector/azure/graph/src/main/java/com/blazebit/query/connector/azure/graph/ManagedDeviceDataFetcher.java

```diff
@@ -49,7 +50,17 @@ public List<ManagedDevice> fetch(DataFetchContext context) {
             List<GraphServiceClient> graphClients = AzureGraphConnectorConfig.GRAPH_SERVICE_CLIENT.getAll(context);
             List<ManagedDevice> list = new ArrayList<>();
             for (GraphServiceClient graphClient : graphClients) {
-                list.addAll(graphClient.deviceManagement().managedDevices().get().getValue());
+                try {
+                    list.addAll(graphClient.deviceManagement().managedDevices().get().getValue());
+                } catch (ODataError oDataError) {
+                    // Check for the specific error indicating Intune is not available
+                    if (oDataError.getMessage().contains("AADSTS500014")) {
+                        // If the error is thrown, Intune is not in use: return an empty list
+                        return list;
+                    } else {
+                        throw oDataError;
+                    }
+                }
```

Comment:
We already did, but maybe we should discuss whether it is better to make this distinction during a test instead of here. This should probably just fail.

Reply:
Replaced this with checking if an intune license is available